junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/execution/JupiterEngineExecutionContextTests.java

```diff
@@ -86,4 +89,20 @@ void canOverrideAttributeWhenContextIsExtended() {
 		assertSame(testInstanceProvider, newContext.getTestInstanceProvider());
 	}
 
+	@Test
+	void closeAttemptExceptionWillBeThrownDownTheCallStack() {
+		FailingAutoClosableExtenstionContext failingExtensionContext = mock(FailingAutoClosableExtenstionContext.class);
+		doCallRealMethod().when(failingExtensionContext).close();
```

Comment:
Why use a mock and a real class? I'd prefer just the mock with a stubbed exception in this case.